sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -79,12 +81,273 @@ object GpuHashJoin {
     }
   }
 
-  def incRefCount(cb: ColumnarBatch): ColumnarBatch = {
-    GpuColumnVector.extractBases(cb).foreach(_.incRefCount())
-    cb
+  def extractTopLevelAttributes(
+      exprs: Seq[Expression],
+      includeAlias: Boolean): Seq[Option[Attribute]] =
+    exprs.map {
+      case a: AttributeReference => Some(a.toAttribute)
+      case GpuAlias(a: AttributeReference, _) if includeAlias => Some(a.toAttribute)
+      case _ => None
+    }
+
+  /**
+   * Filter rows from the batch where all of the keys are null.
+   */
+  def filterNulls(cb: ColumnarBatch, boundKeys: Seq[Expression]): ColumnarBatch = {
```

Comment:
Should we move here the comment we had before? Sorry if I've missed it somewhere else:

> // This is a work around added in response to https://github.com/NVIDIA/spark-rapids/issues/1643.
  // to deal with slowness arising from many nulls in the build-side of the join. The work around
  // should be removed when https://github.com/rapidsai/cudf/issues/7300 is addressed.

Reply:
No because it is not just for that issue any longer.  The Struct join support has extended it.